oidc/clientassertion/options.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package clientassertion
+
+import (
+	"crypto/rsa"
+	"errors"
+
+	"github.com/go-jose/go-jose/v4"
+)
+
+// Option configures the JWT
+type Option func(*JWT) error
+
+// WithClientSecret sets a secret and algorithm to sign the JWT with.
+// alg must be one of:
+// * HS256 with a >= 32 byte secret
+// * HS384 with a >= 48 byte secret
+// * HS512 with a >= 64 byte secret
+func WithClientSecret(secret string, alg HSAlgorithm) Option {
+	return func(j *JWT) error {
+		if err := alg.Validate(secret); err != nil {
+			return err
+		}
+		j.secret = secret
+		j.alg = jose.SignatureAlgorithm(alg)
+		return nil
```

Comment:
Maybe this should error if `j.key` is set?

Reply:
I was letting `validate()` handle that, since it's a cross-attribute check, and it's already ensuring that at least one of them is set.